static/js/unread.js

```diff
@@ -500,6 +500,24 @@ exports.process_loaded_messages = function (messages) {
     });
 };
 
+exports.process_updated_messages = function (message) {
+    // We don't need to add the message in unread_messages list in case
+    // of updated messages. If the message is being updated and it is
+    // not read by the client yet, it will be present in unread_messages
+    // list already. So, we only care about unread_mentions_counter.
+
+    if (!unread_messages.has(message.id)) {
+        // Ignore if the message has been read by the user.
+        return;
+    }
+    const is_unmuted_mention = message.type === 'stream' && message.mentioned &&
+                               !muting.is_topic_muted(message.stream_id,
+                                                      util.get_message_topic(message));
+    if (message.mentioned_me_directly || is_unmuted_mention) {
+        exports.unread_mentions_counter.add(message.id);
+    }
+};
```

Comment:
This logic needs an update in case the mention was **removed** in the edit.

Also, the function name isn't right for a function that only processes a single message.

Reply:
Okay, I am working on it.